package-res/reportviewer/reportviewer-prompt.js

```diff
@@ -186,7 +186,16 @@ define([
       updateParameterDefinitionWithNewValues: function(callback, names) {
         var paramDefn = $.extend(true, {}, this._oldParameterDefinition); // clone previous(old) paramDefn
         for(var i=0; i<names.length; i++) {
-          this.api.util.validateSingleParameter(paramDefn, names[i], this.api.operation.getParameterValues()[names[i]], this._defaultValuesMap);
+          /*As far as we don't recieve all parameters from server we can loose
+            calculated formula results. Let's handle this case. */
+          var untrusted = this.api.operation.getParameterValues()[names[i]];
+          if(undefined === untrusted ){
+            var p = paramDefn.getParameter(names[i]);
+            if(p && p.attributes && p.attributes['post-processor-formula']){
```

Comment:
Instead of latching on to the post-processor formula, you should be able to simply test for the "mandatory-server-validation" (or whatever its name) attribute/flag instead. This flag will be set to true if there is any chance that the value returned is not a fixed result (including post-processor formula's potential changes).